src/Microsoft.DotNet.Darc/src/DarcLib/RemoteRepoBase.cs

```diff
@@ -39,87 +39,53 @@ protected RemoteRepoBase(string temporaryRepositoryPath)
-                                    byte[] contentBytes = GetUtf8ContentBytes(file.Content, file.ContentEncoding);
-                                    await stream.WriteAsync(contentBytes, 0, contentBytes.Length);
-                                }
+                        if (file.Operation == GitFileOperation.Add)
+                        {
+                            if (!File.Exists(filePath))
+                            {
+                                string parentFolder = Directory.GetParent(filePath).FullName;
 
-                                LibGit2SharpHelpers.AddFileToIndex(localRepo, file, filePath, _logger);
+                                Directory.CreateDirectory(parentFolder);
                             }
-                            else
+
+                            using (FileStream stream = File.Create(filePath))
                             {
-                                File.Delete(Path.Combine(tempRepoFolder, file.FilePath));
+                                byte[] contentBytes = GetUtf8ContentBytes(file.Content, file.ContentEncoding);
+                                await stream.WriteAsync(contentBytes, 0, contentBytes.Length);
                             }
                         }
-
-                        localRepo.Index.Write();
-
-                        LibGit2Sharp.Signature author = new LibGit2Sharp.Signature(dotnetMaestro, $"@{dotnetMaestro}", DateTime.Now);
-                        LibGit2Sharp.Signature commiter = author;
-                        localRepo.Commit(commitMessage, author, commiter, new LibGit2Sharp.CommitOptions
+                        else if (file.Operation == GitFileOperation.Delete)
                         {
-                            AllowEmptyCommit = false,
-                            PrettifyMessage = true
-                        });
+                            File.Delete(filePath);
+                        }
 
-                        localRepo.Network.Push(localRepo.Branches[branch], new LibGit2Sharp.PushOptions
-                        {
-                            CredentialsProvider = (url, user, cred) =>
-                            new LibGit2Sharp.UsernamePasswordCredentials
-                            {
-                                // The PAT is actually the only thing that matters here, the username
-                                // will be ignored.
-                                Username = dotnetMaestro,
-                                Password = pat
-                            }
-                        });
+                        LocalHelpers.ExecuteCommand("git", $"add {filePath}", _logger, clonedRepo);
                     }
-                }
-                catch (LibGit2Sharp.EmptyCommitException)
-                {
-                    _logger.LogInformation("There was nothing to commit...");
+
+                    LocalHelpers.ExecuteCommand("git", $"commit -m \"{commitMessage}\"", _logger, clonedRepo);
```

Comment:
In the old code we have handling for empty commits.  Is that possible here?  Without --allow-empty the git commit command will fail.

Reply:
The old code had `AllowEmptyCommit = false,` which mimics what we do here since we are not setting `--allow-empty` so I think the commit command is correct. Also, I don't think that we'll have empty commits and in the case we would, it would be ok to fail.

@mmitche does this answer your comment?

Reply:
Which would cause git to return non-zero, which I believe would throw and then get caught appropriately.  Looks good to me.